src/distinct.rs

```diff
@@ -227,13 +182,107 @@ pub fn rearrange_sequence(colors: &mut Vec<Color>, metric: DistanceMetric) {
         DistanceMetric::CIEDE2000 => c1.distance_delta_e_ciede2000(c2),
     };
 
+    // vector where the i-th element contains the minimum distance to the colors from 0 to i-1.
```

Comment:
Should this be "contains the minimum distance to the colors from i+1 to n"?

Reply:
I think it's correct the way it is. The element at position i is the minimum distance to the current left part of the colors which is 0..i . Note that I've inverted a bit the previous logic where it was using `-` and `max` to take the minimum of the distances to the color and then sorting in descending order. Hopefully now this logic should be a bit more clear and matches the description of the algorithm.